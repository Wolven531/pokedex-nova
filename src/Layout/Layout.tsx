import React, { FC } from 'react'
import { Outlet } from 'react-router-dom'
import { Nav } from '../Nav'

export interface ILayoutProps {}

export const Layout: FC<ILayoutProps> = () => {
	return (
		<>
			<header>
				<h2>Pokedex Nova</h2>
			</header>
			<Nav />
			<section>
				<Outlet />
			</section>
			<footer>&copy; Anthony Williams 2022</footer>
		</>
	)
}
