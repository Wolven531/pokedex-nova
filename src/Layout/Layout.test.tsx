import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { render, RenderResult } from '@testing-library/react'
import { Layout } from './Layout'

describe('Layout', () => {
	let comp: RenderResult

	beforeEach(() => {
		comp = render(
			<MemoryRouter initialEntries={['/']} initialIndex={0}>
				<Layout />
			</MemoryRouter>,
		)
	})

	it('matches snapshot', () => {
		expect(comp.asFragment()).toMatchSnapshot()
	})
})
