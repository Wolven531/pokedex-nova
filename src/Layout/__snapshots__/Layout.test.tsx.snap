// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <h2>
      Pokedex Nova
    </h2>
  </header>
  <nav>
    <ul>
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/other"
        >
          Other
        </a>
      </li>
    </ul>
  </nav>
  <section />
  <footer>
    Â© Anthony Williams 2022
  </footer>
</DocumentFragment>
`;
